pkg/internal/util/kustomize/kutomize_test.go

```diff
@@ -0,0 +1,216 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package kustomize
+
+import (
+	"testing"
+
+	"sigs.k8s.io/kind/pkg/internal/apis/config"
+	"sigs.k8s.io/kind/pkg/internal/util/assert"
+)
+
+func TestBuild(t *testing.T) {
+	type testCase struct {
+		Name            string
+		Resources       []string
+		Patches         []string
+		PatchesJSON6902 []config.PatchJSON6902
+		ExpectError     bool
+		ExpectOutput    string
+	}
+	cases := []testCase{
+		{
+			Name:         "kubeadm config no patches",
+			Resources:    []string{normalKubeadmConfig},
+			ExpectError:  false,
+			ExpectOutput: normalKubeadmConfigKustomized,
+		},
+		{
+			Name:        "kubeadm config bogus patches",
+			Resources:   []string{normalKubeadmConfig},
+			Patches:     []string{"b o g u s"},
+			ExpectError: true,
+		},
+	}
+	for _, tc := range cases {
+		tc := tc // capture test case
+		t.Run(tc.Name, func(t *testing.T) {
+			t.Parallel()
+			out, err := Build(tc.Resources, tc.Patches, tc.PatchesJSON6902)
+			assert.ExpectError(t, tc.ExpectError, err)
```

Comment:
ExpectError contains Errorf, but should it be Fatalf, so that the test exits right away?
in this case it's fine as `out` is a string that is later compared.

perhaps ExpectError should have an optional `bool fatal` flag.

Reply:
I didn't need either of those behaviors currently, if we need it later we can adapt it, otherwise YAGNI

Reply:
continuing on is often preferable to produce more information about other things that do not check out.

Reply:
ok, sgtm.